lib/bolt/cli.rb

```diff
@@ -24,14 +25,15 @@
 require 'bolt/project_manager'
 require 'bolt/puppetdb'
 require 'bolt/rerun'
-require 'bolt/secret'
 require 'bolt/target'
 require 'bolt/version'
 
 module Bolt
   class CLIExit < StandardError; end
 
   class CLI
+    attr_reader :outputter, :rerun
```

Comment:
Is the outputter just for testing? I'm not sure where rerun gets called?

Reply:
Both are set as instance variables in `#execute` and the readers are called in `#process_command`. Mostly set these as instance variables since they're specific to the CLI, but we can pass them as parameters to `#process_command` and get rid of the readers.